Feature/Optional Susy (and other libraries)

cmalven opened the pull request:
Closes #103 

Updates all `src/styles` template to conditionally include code only related to the plugins actually installed. For instance, if you choose not to install the default `susy`, `sass-mq`, or `one-sass-toolkit` plugins, now you won't end up with any code depending on those plugins.

cmalven commented on generators/app/modules/prompts.js:
```diff
@@ -87,7 +87,7 @@ module.exports = [
       {
         name: 'susy',
-        value: 'susy',
+        value: 'susy@2.2.12',
```
I snuck this one in here because its somewhat related and important.

The latest version of Susy is 3.x.

Susy 3.x is an entirely different library from Susy 2.x. It technically does similar things, but in a way that is so different and requires such different setup and philosophy that they might as well be two completely different libraries. You can [read more about it in this article](http://oddbird.net/2017/06/28/susy3/)

Despite 2.x and 3.x being so different, `yarn add susy` now installs 3.x automatically. If you haven't run into this yet and been confused when your susy code doesn't work, you probably will be eventually. Susy 3 completely eliminates any float, block, or flexbased grid compatibility, instead relying on you to implement your own underlying code for these things. Susy 3 is, basically, a "grid math" library and nothing more.

We can have a debate about whether we even want to include susy by default in this generator. Maybe we don't? I still use it on every project so I would vote "yes". However, if we are going to continue to include it by default, I think we should automatically be including the last available 2.x version (which this commit does) until we have an alternative grid strategy in place.

brianjhanson replied:
I've barely used Susy on my last few projects finding it less necessary with flexbox and grid. At first glance, it looks like I would prefer Susy 3 to Susy 2 (since the span math is pretty much all I would need from it) but I'm not sure how other people have been using it.

cc: @collinjoyce @MikeMcChillin @molwells

cmalven replied:
As a point of reference, I went back through the last handful of sites ODC has built to check for common use of Susy 2 mixins (`span()` being the most common), and here's what I found:

**Yes** means Susy 2 mixins were used, **No** means they weren't

```
Freeman:       Yes
Field Notes:   Yes
Hines:         Yes
Sight Machine: Yes
Realtor:       No
Haute Living:  Yes
JLLIPT:        Yes
Tend:          Yes
Uptake:        Yes
Fleetguide:    No
Belgravia:     No
Ghafari:       Yes
```

I think the determining factor for most of these was probably the primary dev on the project, but still interesting.

cmalven replied:
Another option here would be that the generator doesn't default to installing `susy`, but gives you the option of installing either `susy 2` or `susy 3` depending on how you want to handle grids for your project.

molwells replied:
I like that ^^ option @cmalven. I don't use Suzy much, so I'm fine with not installing it by default.

cmalven commented:
I just updated this to not install Susy by default. Since it's questionable whether anyone would use Susy 3 even if we added the ability to install it, I'm not going to bother adding it.
